### PR TITLE
[Deen Liong] iP

### PR DESCRIPTION
**Pull request by deenliong**

### CODE REVIEW

File: src/main/java/Duke.java [hunk 1]
```diff
@@ -1,10 +1,120 @@
...
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String line;
+        Scanner in = new Scanner(System.in);
+
+        Task[] tasks = new Task[100];
```
Comment (karthikstar) [inline src/main/java/Duke.java]: Please avoid using magic literals.

File: src/main/java/Task.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        if (isDone) { // mark done task with X
```
Comment (karthikstar) [inline src/main/java/Task.java]: Avoid putting the comment on same line as this statement. It should be placed a line above, be indented relative to their position of the code statement if(isDone).

File: src/main/java/Task.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            return "[X] ";
+        } else {
+            return "[ ] ";
+        }
+    }
+
+    public String getDescription() {
```
Comment (karthikstar) [inline src/main/java/Task.java]: Good naming convention used. Methods have been well written in camel casing.

File: src/main/java/Task.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        } else {
+            this.isDone = false;
+        }
+    }
+
+    @Override
+    public String toString() {
+
+        return getStatusIcon() + description;
+//        return getStatusIcon() ;
```
Comment (karthikstar) [inline src/main/java/Task.java]: Remove unncessary comments like this.

File: src/main/java/Duke.java [hunk 5]
```diff
@@ -1,10 +1,120 @@
+
+import java.util.Scanner;
+
 public class Duke {
```
Comment (karthikstar) [inline src/main/java/Duke.java]: Class namings look good.

File: src/main/java/Duke.java [hunk 6]
```diff
@@ -1,10 +1,120 @@
+
+import java.util.Scanner;
```
Comment (karthikstar) [inline src/main/java/Duke.java]: Explicit listing of imported classes. Good!

File: src/main/java/Duke.java [hunk 7]
```diff
@@ -1,10 +1,120 @@
...
+    public static String getInputDetails(String line) {
+        String breakLine[] = line.split(" ", 2);
+        return breakLine[1];
+    }
+
+    public static void printTaskList(Task[] tasks, int taskCount) {
+        System.out.println("\t_____________________");
+        // if there are no task
+        if (taskCount == 0) {
+            System.out.println("There are no tasks yet!");
```
Comment (karthikstar) [inline src/main/java/Duke.java]: Conditionals placed on seperate lines and wrapped with curly braces, even if they are single statement ones. Good!

**Review by karthikstar — COMMENTED** [review]
Good work so far, here are some of my comments for your consideration :)

File: src/main/java/Duke.java (hunk 1 above)
Comment (OVReader) [reply src/main/java/Duke.java]: Please avoid magic literal
Suggest: Use static final

File: src/main/java/Todo.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+public class Todo extends Task {
+
+    public Todo(String description) {
+        super(description);
+    }
+
+    @Override
+    public String toString() {
+
```
Comment (OVReader) [inline src/main/java/Todo.java]: Try to avoid white space for more compact code

File: src/main/java/Task.java (hunk 4 above)
Comment (OVReader) [reply src/main/java/Task.java]: Try to avoid redundant comments

File: src/main/java/Todo.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+public class Todo extends Task {
+
```
Comment (OVReader) [inline src/main/java/Todo.java]: Try to avoid white space

**Review by OVReader — APPROVED** [review]
Great Coding Quality! Just some minor code cleanup required

**Pull request closed without merging**